core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/MultipleChoiceEditorDirective.js

```diff
@@ -0,0 +1,162 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the a multiple choice question in the simple
+ * editor.
+ */
+
+oppia.directive('multipleChoiceEditor', [function() {
```

Comment:
Looking at this makes me wonder whether we should just go ahead and add some of the domain objects that are part of the exploration structure to the frontend. I'm thinking at least ``AnswerGroupObjectFactoryService`` and ``OutcomeObjectFactoryService``. It would greatly clean up a lot of this code.

I'm having a tough time understanding exactly what the structure of a 'simple exploration' is, too, since it's superimposed on regular explorations. Because of that, it's hard to suggest what sort of logic should be in a service versus directives. With collections, it was clear. We had data objects like ``CollectionNodeObjectFactory``, we needed to display a list of those so we had ``CollectionNodeListDirective``, and any major changes were done directly to the object factory or through higher-level data transformation services like ``CollectionLinearizerService``. I think that fits MVC quite well, but it's hard to apply principles like that here when considering the existing exploration editor architecture.